pygpudrive/env/env_torch.py

```diff
@@ -249,21 +245,52 @@ def normalize_and_flatten_partner_obs(self, obs):
         obs[:, :, :, 4] /= self.config.max_veh_len
         obs[:, :, :, 5] /= self.config.max_veh_width
 
-        # Object type
-        shifted_type_obs = obs[:, :, :, 6] - 6
+        # One-hot encode the type of the other visible objects
+        one_hot_encoded_object_types = self.one_hot_encode_object_type(obs[:, :, :, 6])
+        
+        # Concat the one-hot encoding with the rest of the features
+        obs = torch.concat((obs[:, :, :, :6], one_hot_encoded_object_types), dim=-1)
+
+        return obs.flatten(start_dim=2)
+    
+    def one_hot_encode_roadpoints(self, roadmap_type_tensor):
+
+        # Set garbage object types to zero
+        road_types = torch.where(
+            (roadmap_type_tensor < self.MIN_OBJ_ENTITY_ENUM) | (roadmap_type_tensor > self.ROAD_MAP_OBJECT_TYPES),
+            0.0,
+            roadmap_type_tensor,
+        ).int()
+        
+        return torch.nn.functional.one_hot(
+                road_types.long(), num_classes=self.ROAD_MAP_OBJECT_TYPES,
+            )
+
+    def one_hot_encode_object_type(self, object_type_tensor):
+        """One-hot encode the object type."""
+        
+        VEHICLE = self.ENTITY_TYPE_TO_INT[gpudrive.EntityType.Vehicle]
+        PEDESTRIAN = self.ENTITY_TYPE_TO_INT[gpudrive.EntityType.Pedestrian]
+        CYCLIST = self.ENTITY_TYPE_TO_INT[gpudrive.EntityType.Cyclist]
+        PADDING = self.ENTITY_TYPE_TO_INT[gpudrive.EntityType._None]
```

Comment:
`gpudrive.EntityType._None` is a different `EntityType` than `gpudrive.EntityType.Padding`. Should one be remapped to the other?

Reply:
Ah, that's a bug!

Reply:
TODO: Add None_ -> there are 5 types